src/riot-rs-threads/src/thread.rs

```diff
@@ -4,6 +4,10 @@ use crate::{thread_flags::ThreadFlags, Arch, Cpu, RunqueueId, ThreadData, Thread
 #[derive(Debug)]
 pub struct Thread {
     /// Saved stack pointer after context switch.
+    #[expect(
+        dead_code,
+        reason = "sp is used in context-specific scheduler implementation"
+    )]
```

Comment:
Ahh you're right. Sorry, I should noticed that `expect` would cause a warning as soon as we compile for `cortex_m` or `riscv`, where the `sp` is used. Opened #387.